### ISSUE TITLE
Unix: Unnecessary copies for promoted struct arguments

### ISSUE THREAD

**Issue by sejongoh** [dotnet/coreclr#6266]
If an struct argument is promoted to registers, the codegen cannot handle this and copies should be generated in `fgMorphArgs`. It is desired to avoid copies and make the codegen copy struct directly to outgoing argument area.

Disabling the following code in `fgMorphArgs' can reproduce the problem.

```
if (lclVar != NULL)
{
    // If the struct is promoted to registers, it has to be materialized
    // on stack. We may want to support promoted structures in
    // codegening pugarg_stk instead of creating a copy here.
    LclVarDsc*  varDsc = &lvaTable[lclVar->gtLclVarCommon.gtLclNum];
    needCpyBlk = varDsc->lvPromoted;
}
```

**Comment by sejongoh** [comment]
Original issue: #2908

**Comment by CarolEidt** [comment]
Closed with #18358
